src/lambda_handler.py

```diff
@@ -1,89 +1,215 @@
+        version = "V3" if "baseMetricV3" in cve["impact"] else "V2"
+        cvss_base_score = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "baseScore"
+        ]
+        cvss_version_temp = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "version"
+        ]
+
+        # Fill document fields with CVE data
+        print(".", end="")
+
+        print("Adding CVE: ", cve_id, " to database")
+
+        cve_doc = CVEDoc(
+            id=cve_id, cvss_score=float(cvss_base_score), cvss_version=cvss_version_temp
         )
-        result["message"] = cowsay.get_output_string(character, contents)
 
-    return result
+        # Gather all saving tasks to run concurrently
+        tasks = [asyncio.create_task(cve_doc.save())]
+        await asyncio.gather(*tasks)
 
+        return cve_doc.id
 
-def task_divide(event) -> dict[str, Union[Optional[float], bool]]:
-    """Divide one number by another and provide the result."""
-    result: dict[str, Union[Optional[float], bool]] = {"result": None, "success": True}
-    numerator: str = event.get("numerator", None)
-    denominator: str = event.get("denominator", None)
 
-    if denominator is None or numerator is None:
-        error_msg = "Request must include both a numerator and a denominator."
-        logging.error(error_msg)
-        failed_task(result, error_msg)
-    else:
-        try:
-            variable_error_msg = "numerator: %s, denominator: %s"
-            result["result"] = example_div(int(numerator), int(denominator))
-        except ValueError:
-            error_msg = "The provided values must be integers."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-        except ZeroDivisionError:
-            error_msg = "The denominator cannot be zero."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-
-    return result
-
-
-def handler(event, context) -> dict[str, Optional[str]]:
+async def process_cve_json(json_stream) -> None:
+    """Process the provided CVEs JSONs and update the database with their contents."""
+    data = json.load(json_stream)
+
+    if data.get("CVE_data_type") != "CVE":
+        raise ValueError("JSON does not look like valid NVD CVE data.")
+
+    tasks = [asyncio.create_task(process_nvd(cve)) for cve in data.get("CVE_Items", [])]
+    await asyncio.gather(*tasks)
+
+
+async def unzip_file(socket) -> None:
+    """With the open socket, unzip the file and process it with process_cve_json()."""
+    # Open the file and read it into memory
+    buf = BytesIO(socket.read())
+    f = gzip.GzipFile(fileobj=buf)
+    await process_cve_json(f)
+
+
+async def process_urls(cve_urls, db) -> None:
+    """Initialize beanie ODM and begin processing CVE URLs."""
```

Comment:
Use proper capitalization of package name:
```suggestion
    """Initialize Beanie ODM and begin processing CVE URLs."""
```

Reply:
@dav3r I addressed this here [Line 154](https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/e6397c680698de8835b0629e200b28fe2829df5b#diff-4655711b74e9f79a2566fdbac7451f4dcc265a12d3ad743e519353283898a155R154)